sdk/couchbase-core/src/mgmtx/bucket_settings.rs

```diff
@@ -0,0 +1,515 @@
+        }
+    }
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub(crate) enum InnerStorageBackend {
+    Couchstore,
+    Magma,
+    Other(String),
+}
+
+pub(crate) fn encode_mutable_bucket_settings(
+    serializer: &mut Serializer<String>,
+    opts: &MutableBucketSettings,
+) {
+    if let Some(flush) = opts.flush_enabled {
+        serializer.append_pair("flushEnabled", if flush { "1" } else { "0" });
+    }
+    if let Some(quota) = opts.ram_quota_mb {
+        serializer.append_pair("ramQuotaMB", quota.to_string().as_str());
+    }
+    if let Some(num) = opts.replica_number {
+        serializer.append_pair("replicaNumber", num.to_string().as_str());
+    }
+    if let Some(eviction_policy) = &opts.eviction_policy {
+        serializer.append_pair("evictionPolicy", eviction_policy.to_string().as_str());
+    }
+    if let Some(max_ttl) = &opts.max_ttl {
+        serializer.append_pair("maxTTL", max_ttl.as_secs().to_string().as_str());
+    }
+    if let Some(compression_mode) = &opts.compression_mode {
+        serializer.append_pair("compressionMode", compression_mode.to_string().as_str());
+    }
+    if let Some(durability_min_level) = &opts.durability_min_level {
+        serializer.append_pair(
+            "durabilityMinLevel",
+            durability_min_level.to_string().as_str(),
+        );
+    }
+    if let Some(retention) = opts.history_retention_bytes {
+        serializer.append_pair("historyRetentionBytes", retention.to_string().as_str());
+    }
+    if let Some(retention) = opts.history_retention_seconds {
+        serializer.append_pair("historyRetentionSeconds", retention.to_string().as_str());
+    }
+    if let Some(history_retention_collection_default) = &opts.history_retention_collection_default {
+        serializer.append_pair(
+            "historyRetentionCollectionDefault",
+            history_retention_collection_default.to_string().as_str(),
+        );
+    }
+}
+
+pub(crate) fn encode_bucket_settings(
+    serializer: &mut Serializer<String>,
+    opts: &BucketSettings,
+) -> error::Result<()> {
+    encode_mutable_bucket_settings(serializer, &opts.mutable_bucket_settings);
+
+    if let Some(conflict_resolution_type) = &opts.conflict_resolution_type {
+        serializer.append_pair(
+            "conflictResolutionType",
+            conflict_resolution_type.to_string().as_str(),
+        );
+    }
+    if let Some(index) = opts.replica_index {
+        serializer.append_pair("replicaIndex", if index { "1" } else { "0" });
+    }
+    if let Some(bucket_type) = &opts.bucket_type {
+        serializer.append_pair("bucketType", bucket_type.to_string().as_str());
+    }
+    if let Some(storage_backend) = &opts.storage_backend {
+        serializer.append_pair("storageBackend", storage_backend.to_string().as_str());
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_bucket_type() {
```

Comment:
What is the point of this test?

Reply:
I was using it test out the feel of the pattern we've used instead of an enum, deleted.